stories/s5pl2.stories.js

```diff
@@ -0,0 +1,320 @@
+    const getMapParams = {
+      bbox: bbox,
+      fromTime: new Date(Date.UTC(2020, 2 - 1, 2, 0, 0, 0)),
+      toTime: new Date(Date.UTC(2020, 2 - 1, 2, 23, 59, 59)),
+      width: 512,
+      height: 512,
+      format: MimeTypes.JPEG,
+    };
+    const imageBlob = await layer.getMap(getMapParams, ApiType.WMS);
+    img.src = URL.createObjectURL(imageBlob);
+  };
+  perform().then(() => {});
+
+  return wrapperEl;
+};
+
+export const getMapProcessing = () => {
+  if (!process.env.CLIENT_ID || !process.env.CLIENT_SECRET) {
+    return "<div>Please set OAuth Client's id and secret for Processing API (CLIENT_ID, CLIENT_SECRET env vars)</div>";
+  }
+
+  const img = document.createElement('img');
+  img.width = '512';
+  img.height = '512';
+
+  const wrapperEl = document.createElement('div');
+  wrapperEl.innerHTML = "<h2>GetMap with Processing</h2>";
+  wrapperEl.insertAdjacentElement("beforeend", img);
+
+  const perform = async () => {
+    await setAuthTokenWithOAuthCredentials();
+
+    const layer = new S5PL2Layer(
+      instanceId,
+      layerId,
+      `
+      //VERSION=3
+      function setup() {
+        return {
+          input: ["CO"],
+          output: { bands: 3 }
+        };
+      }
+
+      function evaluatePixel(sample) {
+        return [2.5 * sample.CO, 2.5 * sample.CO, 2.5 * sample.CO];
+      }
+    `,
+    );
+
+    const bbox = new BBox(CRS_EPSG3857, -2035059.4, 15497760.4, -1956787.9, 15576031.8);
+    const getMapParams = {
+      bbox: bbox,
+      fromTime: new Date(Date.UTC(2020, 2 - 1, 2, 0, 0, 0)),
+      toTime: new Date(Date.UTC(2020, 2 - 1, 2, 23, 59, 59)),
+      width: 512,
+      height: 512,
+      format: MimeTypes.JPEG,
+      maxCCPercent: 100,
+    };
+    const imageBlob = await layer.getMap(getMapParams, ApiType.PROCESSING);
+    img.src = URL.createObjectURL(imageBlob);
+  };
+  perform().then(() => {});
+
+  return wrapperEl;
+};
+
+export const getMapProcessingWithoutInstance = () => {
+  if (!process.env.CLIENT_ID || !process.env.CLIENT_SECRET) {
+    return "<div>Please set OAuth Client's id and secret for Processing API (CLIENT_ID, CLIENT_SECRET env vars)</div>";
+  }
+
+  const img = document.createElement('img');
+  img.width = '512';
+  img.height = '512';
+
+  const wrapperEl = document.createElement('div');
+  wrapperEl.innerHTML = "<h2>GetMap with Processing</h2>";
+  wrapperEl.insertAdjacentElement("beforeend", img);
+
+  const perform = async () => {
+    await setAuthTokenWithOAuthCredentials();
+
+    const layer = new S5PL2Layer(
+      null,
+      null,
+      `
+      //VERSION=3
+      function setup() {
+        return {
+          input: ["CO"],
+          output: { bands: 3 }
+        };
+      }
+
+      function evaluatePixel(sample) {
+        return [2.5 * sample.CO, 2.5 * sample.CO, 2.5 * sample.CO];
+      }
+    `,null, null, null, null, AcquisitionMode.EW, Polarization.DH, Resolution.MEDIUM
```

Comment:
These are not valid S5 options (S1 leftovers?)

Reply:
Oops, thank you, fixed. :)